lib/request_handler/schema_parser.rb

```diff
@@ -17,6 +17,7 @@ def initialize(schema:, schema_options: {})
 
     def validate_schema(data, with: schema)
       raise MissingArgumentError, data: 'is missing' if data.nil?
+      data = deep_symbolize_keys(data) if with.rules.keys.first.is_a?(Symbol)
```

Comment:
for the usecase where I want to update NULL-values in a nested hash, this somehow only works when the keys are symbols.. tried it out directly with a very simple dry schema and with my complex usecases.

Reply:
so should we then always symbolize the data?
what's the reasoning behind looking at something in the first rule

and what exactly is not working the nullable relationships or nested data in general?

Reply:
we could, yes.. just thought you had a reason to also support strings.
only the nullable relationships were not working before

Reply:
No particular reason just that we had no reason not to support strings.
I mainly want to know why this would impact the nullable relationships and not other nested data.

If I remove this line the tests still work so I do not understand the need for this. Please add add test which demonstrates the problem

Reply:
Sorry I confused this.
This impacts nested hashes, as dry-validation can't handle this properly when the schema is defined with symbols, but we send nested hashes with string-keys there. That's why I implemented deep_symbolize_keys.
I added a test for this and you can also take a look here: https://gist.github.com/sarahdiendorfer/7ef40b6ac4a97082da266249a0b4d68f

Reply:
:+1: ok thanks now it's clear what this is/was about